CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+
+
+### Style
+
+Style is of course the first guideline on every new contributor's mind :)
+
+This codebase largely complies with the [Google C++ style guide](https://google.github.io/styleguide/cppguide.html). The maintainers recommend the use of [Clang's auto formatter](https://clang.llvm.org/docs/ClangFormatStyleOptions.html).
+
+Notable exceptions:
+ 1. C++ exceptions are allowed (in fact, only `lczero::Exception`, defined in `utils/exception.h`, is allowed)
+ 2. We use `#pragma once` instead of header guards.
+ 3. Default function parameters are sometimes allowed, not only for constructors and assignment operators.
+ 4. Rvalue reference function params are sometimes allowed.
+
+For items (3) and (4), usage of those are discouraged, only use them if they benefit readability or have significant performance gain. It's possible that those exceptions (3) and (4) will be disallowed in future.
+
+The most important rule to follow is consistency: look at the surrounding code when doing changes and follow similar style.
+
+These are the most important parts of the codebase style (as a sort of tl;dr):
+
+ * Comments must be full sentences, i.e. capitalized and ending in a period. (Sentences with elided subjects are fine.) Only `//` style comments are allowed, `/* */` style comments aren't.
+
+ * Braces are a variant of K&R style, as can be gleaned from existing code. All `if` statements must use braces, with the possible exception of single statement `if`s, which *may* omit if the braces *if* the conditional and following statement are on the same line. Again, see surrounding code for examples.
+
+ * Indentation is two spaces; \t characters are disallowed.
+
+ * Code line length is strictly capped at 80 characters.
+
+ * Using non-`const` references as function parameters is disallowed; use pointers instead. (Using `const` references as parameters is fine.)
+
+ * Identifier style:
+   - `kLikeThis` for constants and enum values
+   - `like_this` for variables
+   - `like_this_` for member variables
+   - `LikeThis` for function and class names
+
+ * All code should be inside `namespace lczero`
+
+The internal code dependency structure looks like this:
+
+ * Code in `src/utils` is not allowed to depend on any other code.
+
+ * Code in `src/chess` only depends on `src/utils`
+
+ * Code in `src/neural` only depends on `src/utils` and `src/chess`
+
+ * Code in `src/mcts` only depends on `src/utils`, `src/chess` and `src/neural`
+
+
+### Git history
+
+Pull Requests are squahsed when merged. This means all commits in the branch will be squashed into one commit applied onto master, so PRs (and branches) should stick to *one* topic only. If you think changes deserve separate commits, make separate PRs for each commit. (This also means it's not possible to reuse one branch for multiple PRs; new PRs must either use entirely new branches, or else you could use `git reset --hard` on the current branch.)
```

Comment:
`(This also means it's not possible to reuse...` was related to squashing, but now that comment is too far from mentioning of squashing, so it's not clear.
I suggest to replace `This` with `Squashing`, and remove parentheses around that sentence.

Reply:
I had it in parentheses because that sentence isn't really related to our codebase standards, it's only a practical tip for people who understand git very little (which admittedly once included me).